strawberry/experimental/pydantic/object_type.py

```diff
@@ -147,10 +149,17 @@ def wrap(cls):
 
         sorted_fields = missing_default + has_default
 
+        # Implicitly define `is_type_of` to support interfaces/unions that use
+        # pydantic objects (not the corresponding strawberry type)
+        @classmethod  # type: ignore
```

Comment:
Unfortunately, `# type: ignore` is the suggested solution here: https://github.com/python/mypy/issues/4153